internal/extension/smartagentextension/factory.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package smartagentextension
+
+import (
+	"context"
+	"os"
+
+	"github.com/signalfx/signalfx-agent/pkg/core/common/constants"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/extension/extensionhelper"
+)
+
+const (
+	typeStr configmodels.Type = "smartagent"
+)
+
+func NewFactory() component.ExtensionFactory {
+	return extensionhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		createExtension)
+}
+
+func createDefaultConfig() configmodels.Extension {
+	return &Config{
+		ExtensionSettings: configmodels.ExtensionSettings{
+			TypeVal: typeStr,
+			NameVal: string(typeStr),
+		},
+		BundleDir: os.Getenv(constants.BundleDirEnvVar),
```

Comment:
Would it be preferable to adopt agent fallback behavior if env var not set? https://github.com/signalfx/signalfx-agent/blob/master/pkg/core/config/config.go#L181

Reply:
Added the same behavior as in the agent.